### PR TITLE
Prefix config file with its type

### LINKED ISSUE #836: import error in 19.1.0
I've the following error:

    Error: relative imports require the 'package' argument

No such error on previous installed version (19.0.0).

### PR DESCRIPTION
**Pull request by slava-sh**
Fixes #836 as described in @benoitc's [comment](https://github.com/benoitc/gunicorn/issues/836#issuecomment-50957348).

### CODE REVIEW

File: tests/test_003-config.py [hunk 1]
```diff
@@ -186,8 +186,15 @@ def test_load_config():
...
+def test_load_config_explicit_file():
+    with AltArgs(["prog_name", "-c", "file:%s" % cfg_file()]):
+        app = NoConfigApp()
+    t.eq(app.cfg.bind, ["unix:/tmp/bar/baz"])
+    t.eq(app.cfg.workers, 3)
+    t.eq(app.cfg.proc_name, "fooey")
+
 def test_load_config_module():
-    with AltArgs(["prog_name", "-c", cfg_module()]):
+    with AltArgs(["prog_name", "-c", "python:%s" % cfg_module()]):
```
Comment (berkerpeksag) [inline tests/test_003-config.py]: I think it would be nice to keep the test untouched to make sure that we don't break backwards compatibility. We can add a separate test for the `python:` prefix.

File: gunicorn/config.py [hunk 2]
```diff
@@ -465,11 +465,13 @@ class ConfigFile(Setting):
...
     cli = ["-c", "--config"]
-    meta = "FILE"
+    meta = "CONFIG"
     validator = validate_string
     default = None
     desc = """\
-        The path to a Gunicorn config file, or python module.
+        The Gunicorn config file.
+
+        A string of the form 'PATH', 'file:PATH', or 'python:MODULE_NAME'.
```
Comment (berkerpeksag) [inline gunicorn/config.py]: Can you replace the single quotes with backticks(e.g. 'PATH' -> \`\`PATH\`\`)?

Comment (berkerpeksag) [inline gunicorn/config.py]: We can also add a "versionadded" directive (or we can add a note to docs/source/news.rst)

File: docs/source/settings.rst [hunk 3]
```diff
@@ -310,7 +312,7 @@ chdir
 ~~~~~
 
 * ``--chdir``
-* ``/Users/benoitc/work/gunicorn/py27/gunicorn/docs``
+* ``/Users/slava/Desktop/gunicorn/docs``
```
Comment (berkerpeksag) [inline docs/source/settings.rst]: This looks like an unrelated change(and caused by `make -C docs html`) to me. Can you revert it? :)

File: docs/source/settings.rst [hunk 4]
```diff
@@ -413,7 +415,7 @@ temporary directory.
 secure_scheme_headers
 ~~~~~~~~~~~~~~~~~~~~~
 
-* ``{'X-FORWARDED-PROTOCOL': 'ssl', 'X-FORWARDED-PROTO': 'https', 'X-FORWARDED-SSL': 'on'}``
+* ``{'X-FORWARDED-PROTOCOL': 'ssl', 'X-FORWARDED-SSL': 'on', 'X-FORWARDED-PROTO': 'https'}``
```
Comment (berkerpeksag) [inline docs/source/settings.rst]: Same as above.

**Comment by berkerpeksag** [comment]
Thanks for the PR, @slava-sh :)

Three comments:

* Maybe we can also add `py:` as an alternative to `python:`.
* Can you please squash the commits into one when you're done?
* An additional "thank you" for adding tests :)

File: tests/test_003-config.py (hunk 1 above)
Comment (slava-sh) [inline tests/test_003-config.py]: From my understanding of your [comment](https://github.com/benoitc/gunicorn/issues/836#issuecomment-50957348), we do want to break backward compatibility here in order to separate the two cases. What do you think?

File: gunicorn/config.py (hunk 2 above)
Comment (slava-sh) [inline gunicorn/config.py]: Okay. Note that other descriptions, such as [bind](https://github.com/benoitc/gunicorn/blob/72a8e2940d75b3ef7a92edcb3e4247f43426706f/gunicorn/config.py#L494) and [syslog_addr](https://github.com/benoitc/gunicorn/blob/72a8e2940d75b3ef7a92edcb3e4247f43426706f/gunicorn/config.py#L1157), use single quotes as well.

**Comment by slava-sh** [comment]
I don't see a point in having two different prefixes for doing one thing. We can choose a prefix different from `python:` though.

**Comment by tilgovi** [comment]
:+1:

No opinion on `python` vs `py`. Both are fine. We should pick one.

**Comment by benoitc** [comment]
@berkerpeksag why `py` instead of `python`? Shorter?

**Comment by slava-sh** [comment]
How about `module`?

**Comment by tilgovi** [comment]
An alternative idea: use a colon to distinguish between paths and modules, with the colon separating the package from the relative path.

Colon is uncommon in filenames because it's not very portable, so it's likely unambiguous.

Pyramid does this and calls them asset specifications.

"foo/bar" is a relative path
"/path/to/foo/bar" is an absolute path
"foo:bar" is the "bar" path within the "foo" package.

**Comment by benoitc** [comment]
@tilgovi interesting idea. IF the pattern is already used why not. I was just reusing the pattern existing in trac and other stuff like this.

**Comment by berkerpeksag** [comment]
> @berkerpeksag why py instead of python? Shorter?

Yes.

I also like @tilgovi's suggestion since it's already being used in the wild.

**Comment by tilgovi** [comment]
I don't have a preference. Just mentioning it. I haven't seen it anywhere other than Pyramid. If Trac and others have `file:` that's fine, too. Paste uses `config:` and `egg:`. Anything is acceptable.

**Comment by tilgovi** [comment]
I think I like `file:` and `python:` (or `module:`) since there is no guessing about the meaning. It is very explicit.

**Comment by benoitc** [comment]
i would go for python: the code already exists :)

**Comment by tilgovi** [comment]
:+1:

**Comment by tilgovi** [comment]
Needs a rebase, though.

**Comment by slava-sh** [comment]
I think this can be merged now.

**Comment by benoitc** [comment]
LGTM

**Comment by berkerpeksag** [comment]
Thanks!
